department-of-veterans-affairs/caseflow
# Kamalam7/appeals 29116 v1

Resolves https://jira.devops.va.gov/browse/APPEALS-29116

# Description
Created scenario-I edge case as following: 
A task is blocking - cancel all children
A task isn't blocking
this task has children and one of them is blocking - cancel all child tasks of this task
this task has children and none are blocking

## Acceptance Criteria
- [x] Dropdown for attorney legacy task will only populate if an active blocking task is on the legacy appeal
After user selects a judge and clicks reassign all active blocking tasks should be cancelled, including all child elements
Completed blocking tasks should not be marked as cancelled

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

1. From the Generators script, run this "bundle exec rake db:generate_legacy_appeals_with_tasks" and select Scenario1edge and generate data.
2. Recreate Scenario-1 
3. Login as an SCM user BVARDUNKLE and go to his Queue
4. Use the URL http://localhost:3000/queue/appeals/LEGACYPH (pickup the VACOLS ID from the seed script generated)
5. In the Case details page, in the Currently active tasks, from the Actions dropdown you can select "Case Movement: Cancel blocking task and advance to judge"
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/46348e2b-6d52-4127-89dd-3f17e8b51865)
6. In the Reassign case page, select the reason and context for cancellation and click on continue
<img width="830" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/13c7c461-3381-4006-9f1b-55dd84d9b47a">
7.From the Confirm cancellation and reassign modal, select the Judge and provide instruction and click on cancel task and reassign
<img width="805" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/fdd9a8d8-6fe3-495a-ae79-73f3c9403720">
8. You can see the case is assigned to a judge in the case details page, currently active task
<img width="859" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/3cc4cd95-900f-4915-8261-9d038c5e214f">
9. Now the task tree has to show as below, cancelling the Parent and Child tasks and assigning Judge task
<img width="944" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/9b8aca88-fede-4592-8339-31ce34744121">



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
